tests/unit/field/test_field3d.cxx

```diff
@@ -810,13 +1081,50 @@ TEST_F(Field3DTest, AddField3DField3D) {
   EXPECT_TRUE(IsField3DEqualBoutReal(c, 3.0));
 }
 
+TEST_F(Field3DTest, AddField3DField3DStagger) {
+  auto backup = mesh->StaggerGrids;
+  mesh->StaggerGrids = true; // Force staggering
+
+  Field3D a, b, c;
+
+  a = 1.0;
+  b = 2.0;
+  c = 3.0;
+
+  a.setLocation(CELL_XLOW);
+  b.setLocation(CELL_CENTRE);
+  c.setLocation(CELL_CENTRE);
+
+  // Throw as two rhs fields at different locations
+#if CHECK > 0
+  EXPECT_THROW(c = a + b, BoutException);
+#else
+  EXPECT_NO_THROW(c = a + b);
+#endif
+
+  // No throw as updates location of a
+  EXPECT_NO_THROW(a = c + b);
```

Comment:
Should we also check the value of `a` is what we expect in these stagger tests?

Reply:
We could, but we don't actually do anything different than in the non-staggered add -- it's just the checking of locations that could be different I think.